### ISSUE TITLE
[BUG] Is data server functional?

### ISSUE THREAD

**Issue by adc0032** [LieberInstitute/recount3#29]
Checking on the recount3 server. The following command was working for me until today:

```{r }
human_projects<- available_projects()
```
is returning:
"Error: 'recount3_url' is not a valid supported URL since it's missing the URL/<organism>/homes_index text file or 'recount3_url' is not an existing directory in your file system."

[download site](https://duffel.rail.bio/recount3) is also returning a 404 message.

**Comment by camdouglas** [comment]
I am getting the same issue when trying to download packages. 

`SRP067636_rse <- recount3::create_rse_manual(
    project = "SRP067636",
    project_home = "data_sources/sra",
    organism = "mouse",
    annotation = "gencode_v23",
    type = "gene"
)`

I have not found any solution for it yet.

**Comment by lixin4306ren** [comment]
Got the same error. recount2 server also didn't work.

**Comment by lcolladotor** [comment]
What version of `recount3` are you using?

**Comment by lcolladotor** [comment]
Vince has seen this with both bioc-release and bioc-devel https://support.bioconductor.org/p/9148272/#9148276.

**Comment by lcolladotor** [comment]
I've relayed this info to the maintainer of duffel as well as the SciServer admins.

**Comment by adc0032** [comment]
> What version of `recount3` are you using?

1.4 (running R 4.1.2 to validate a project in a docker) 


Thank you for checking in on this!

**Comment by msfuentealba** [comment]
I hope they didn't delete the files. I'm wondering if there's a mirror somewhere...

**Comment by ChristopherWilks** [comment]
Just wanted to post that while this is still an issue (recount3 is down), it's not due to any data loss as the filesystem hosting recount3 is fine.  The issue is with the webserver hosting recount3.  Also, there is no public mirror that I'm aware of, though that would be a good thing to have!

**Comment by lcolladotor** [comment]
Hi,

The SciServer (host) admin that can resolve this issue is traveling today. So like @ChristopherWilks pointed out, things will likely get resolved soon.

Best,
Leo

**Comment by lcolladotor** [comment]
We are aware. No need for the bump, thanks.

**Comment by lcolladotor** [comment]
https://twitter.com/BenLangmead/status/1602655981544022018

**Comment by lcolladotor** [comment]
Hi everyone,

Thanks to your reports, @ChristopherWilks's investigative work, several tests by Jordan R from the IDIES at JHU team (the data host) and ultimately some code changes by @nellore, everything is working again as you can see below.

```R
> human_projects <- recount3::available_projects()
2022-12-14 00:41:24 caching file sra.recount_project.MD.gz.
2022-12-14 00:41:24 caching file gtex.recount_project.MD.gz.
2022-12-14 00:41:24 caching file tcga.recount_project.MD.gz.
> dim(human_projects)
[1] 8742    6
> packageVersion("recount3")
[1] ‘1.8.0’
```

Technically https://duffel.rail.bio/recount3/human/homes_index now works again by pointing to a new location than it used to. That's why we designed recount2 and recount3 around duffel, in case we needed to change the ultimate data host location.

Best,
Leo

**Comment by ndaniel** [comment]
Is it down again or is it just me?

**Comment by nellore** [comment]
@ndaniel Seems not unlikely. TLS 1.3 is not supported, it looks like: https://www.cdn77.com/tls-test/result?domain=sciserver.org . Can you please add `-v` to your curl command and paste the output for us? Maybe https://superuser.com/questions/606598/specifying-minor-tls-version-when-using-curl offers a workaround for now, too.

**Comment by lcolladotor** [comment]
@nellore I'm getting this now:

```bash
% curl -v http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
*   Trying 10.173.34.79:80...
* Connected to duffel.rail.bio (10.173.34.79) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Date: Wed, 14 Dec 2022 15:08:40 GMT
< Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips PHP/5.6.37
< Last-Modified: Tue, 03 Jan 2017 17:27:19 GMT
< ETag: "2b7-54533faedda24"
< Accept-Ranges: bytes
< Content-Length: 695
< Content-Type: text/html; charset=UTF-8
<
<!DOCTYPE html>
<html>
<head>
<title>Blocked Site</title>
</head>
<body>
<center><img src="http://blocked.jh.edu/logo.png" alt="JH Logo"</center>
<center><h1>Web Page Blocked</h1></center>
<h3><p align="left">
The website you are attempting to visit is blocked by IT@JH security policies due to malicious activity being detected on the remote website. If you believe you have received this message in error, please create a self-service ticket via the myJH portal <a href="https://my.jh.edu">https://my.jh.edu</a> click the helpdesk icon and click "report a problem" or contact the helpdesk directly at 410-955-HELP (4357) or 410-516-HELP (4357).</p>
</h3>
<?php echo 'test' ?>
</body>
</html>
```

**Comment by nellore** [comment]
`curl -LOv http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz` tends to work for me. Once it was hanging trying to connect to Cloudflare IP. Have not managed to reproduce your block. Maybe it was resolved?

**Comment by lcolladotor** [comment]
Hi Abhi @nellore,

The `-LOv` options work for me as shown below (`-v` only still triggers the IT @ JH block).

```
% curl -LOv http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 10.173.34.79:80...
* Connected to duffel.rail.bio (10.173.34.79) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Date: Wed, 14 Dec 2022 16:58:21 GMT
< Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips PHP/5.6.37
< Last-Modified: Tue, 03 Jan 2017 17:27:19 GMT
< ETag: "2b7-54533faedda24"
< Accept-Ranges: bytes
< Content-Length: 695
< Content-Type: text/html; charset=UTF-8
<
{ [695 bytes data]
100   695  100   695    0     0   3152      0 --:--:-- --:--:-- --:--:--  3423
* Connection #0 to host duffel.rail.bio left intact
```

```
% curl -v http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
*   Trying 10.173.34.79:80...
* Connected to duffel.rail.bio (10.173.34.79) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Date: Wed, 14 Dec 2022 16:58:27 GMT
< Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips PHP/5.6.37
< Last-Modified: Tue, 03 Jan 2017 17:27:19 GMT
< ETag: "2b7-54533faedda24"
< Accept-Ranges: bytes
< Content-Length: 695
< Content-Type: text/html; charset=UTF-8
<
<!DOCTYPE html>
<html>
<head>
<title>Blocked Site</title>
</head>
<body>
<center><img src="http://blocked.jh.edu/logo.png" alt="JH Logo"</center>
<center><h1>Web Page Blocked</h1></center>
<h3><p align="left">
The website you are attempting to visit is blocked by IT@JH security policies due to malicious activity being detected on the remote website. If you believe you have received this message in error, please create a self-service ticket via the myJH portal <a href="https://my.jh.edu">https://my.jh.edu</a> click the helpdesk icon and click "report a problem" or contact the helpdesk directly at 410-955-HELP (4357) or 410-516-HELP (4357).</p>
</h3>
<?php echo 'test' ?>
</body>
</html>

* Connection #0 to host duffel.rail.bio left intact
```

**Comment by lcolladotor** [comment]
Hi everyone & @nellore,

Hmm, setting `recount3_url = "https://sciserver.org/public-data/recount3/data"` (or with `http`) right now works, which avoids the duffel ban by IT @ JH as shown below.

```R
> recount3::project_homes("human", recount3_url = "https://sciserver.org/public-data/recount3/data")
[1] "data_sources/sra"  "data_sources/gtex" "data_sources/tcga"
> recount3::project_homes("human", recount3_url = "http://sciserver.org/public-data/recount3/data")
[1] "data_sources/sra"  "data_sources/gtex" "data_sources/tcga"
>
> recount3::project_homes("human")
Error: 'recount3_url' is not a valid supported URL since it's missing the URL/<organism>/homes_index text file or 'recount3_url' is not an existing directory in your file system.
```

Behind the scenes, `recount3` is using `RCurl::url.exists()` which I can run manually as shown below with the verbose option. 

```R
> RCurl::url.exists("https://sciserver.org/public-data/recount3/data/human/homes_index", .opts = list(verbose = TRUE))
*   Trying 128.220.233.141:443...
* Connected to sciserver.org (128.220.233.141) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server did not agree on a protocol. Uses default.
* Server certificate:
*  subject: C=US; ST=Maryland; O=Johns Hopkins University; CN=sciserver.org
*  start date: Aug 30 00:00:00 2022 GMT
*  expire date: Sep 30 23:59:59 2023 GMT
*  subjectAltName: host "sciserver.org" matched cert's "sciserver.org"
*  issuer: C=US; ST=MI; L=Ann Arbor; O=Internet2; OU=InCommon; CN=InCommon RSA Server CA
*  SSL certificate verify ok.
> HEAD /public-data/recount3/data/human/homes_index HTTP/1.1
Host: sciserver.org
Accept: */*

* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Wed, 14 Dec 2022 17:17:59 GMT
< Server: Apache/2.4.6 (CentOS) PHP/7.3.10
< Last-Modified: Tue, 01 Mar 2022 17:44:39 GMT
< ETag: "35-5d92bb8204c36"
< Accept-Ranges: bytes
< Content-Length: 53
< Cache-Control: max-age=2592000
< Expires: Fri, 13 Jan 2023 17:17:59 GMT
< Set-Cookie: SERVERID=hhpc010; path=/
< Cache-control: private
<
* Connection #0 to host sciserver.org left intact
[1] TRUE
> RCurl::url.exists("https://duffel.rail.bio/recount3/human/homes_index", .opts = list(verbose = TRUE))
*   Trying 10.173.34.79:443...
* connect to 10.173.34.79 port 443 failed: Connection refused
* Failed to connect to duffel.rail.bio port 443 after 121 ms: Connection refused
* Closing connection 0
[1] FALSE
```

I can't figure out how to simulate the `-LO` options with `RCurl::url.exists()` right now though.

```R
> RCurl::listCurlOptions()
  [1] "address.scope"              "append"
  [3] "autoreferer"                "buffersize"
  [5] "cainfo"                     "capath"
  [7] "certinfo"                   "closepolicy"
  [9] "connect.only"               "connecttimeout"
 [11] "connecttimeout.ms"          "conv.from.network.function"
 [13] "conv.from.utf8.function"    "conv.to.network.function"
 [15] "cookie"                     "cookiefile"
 [17] "cookiejar"                  "cookielist"
 [19] "cookiesession"              "copypostfields"
 [21] "crlf"                       "crlfile"
 [23] "customrequest"              "debugdata"
 [25] "debugfunction"              "dirlistonly"
 [27] "dns.cache.timeout"          "dns.use.global.cache"
 [29] "egdsocket"                  "encoding"
 [31] "errorbuffer"                "failonerror"
 [33] "file"                       "filetime"
 [35] "followlocation"             "forbid.reuse"
 [37] "fresh.connect"              "ftp.account"
 [39] "ftp.alternative.to.user"    "ftp.create.missing.dirs"
 [41] "ftp.filemethod"             "ftp.response.timeout"
 [43] "ftp.skip.pasv.ip"           "ftp.ssl"
 [45] "ftp.ssl.ccc"                "ftp.use.eprt"
 [47] "ftp.use.epsv"               "ftpappend"
 [49] "ftplistonly"                "ftpport"
 [51] "ftpsslauth"                 "header"
 [53] "headerfunction"             "http.content.decoding"
 [55] "http.transfer.decoding"     "http.version"
 [57] "http200aliases"             "httpauth"
 [59] "httpget"                    "httpheader"
 [61] "httppost"                   "httpproxytunnel"
 [63] "ignore.content.length"      "infile"
 [65] "infilesize"                 "infilesize.large"
 [67] "interface"                  "ioctldata"
 [69] "ioctlfunction"              "ipresolve"
 [71] "issuercert"                 "keypasswd"
 [73] "krb4level"                  "krblevel"
 [75] "localport"                  "localportrange"
 [77] "low.speed.limit"            "low.speed.time"
 [79] "max.recv.speed.large"       "max.send.speed.large"
 [81] "maxconnects"                "maxfilesize"
 [83] "maxfilesize.large"          "maxredirs"
 [85] "netrc"                      "netrc.file"
 [87] "new.directory.perms"        "new.file.perms"
 [89] "nobody"                     "noprogress"
 [91] "noproxy"                    "nosignal"
 [93] "opensocketdata"             "opensocketfunction"
 [95] "password"                   "port"
 [97] "post"                       "post301"
 [99] "postfields"                 "postfieldsize"
[101] "postfieldsize.large"        "postquote"
[103] "postredir"                  "prequote"
[105] "private"                    "progressdata"
[107] "progressfunction"           "protocols"
[109] "proxy"                      "proxy.transfer.mode"
[111] "proxyauth"                  "proxypassword"
[113] "proxyport"                  "proxytype"
[115] "proxyusername"              "proxyuserpwd"
[117] "put"                        "quote"
[119] "random.file"                "range"
[121] "readfunction"               "redir.protocols"
[123] "referer"                    "resume.from"
[125] "resume.from.large"          "seekdata"
[127] "seekfunction"               "share"
[129] "sockoptdata"                "sockoptfunction"
[131] "socks5.gssapi.nec"          "socks5.gssapi.service"
[133] "ssh.auth.types"             "ssh.host.public.key.md5"
[135] "ssh.private.keyfile"        "ssh.public.keyfile"
[137] "ssl.cipher.list"            "ssl.ctx.data"
[139] "ssl.ctx.function"           "ssl.sessionid.cache"
[141] "ssl.verifyhost"             "ssl.verifypeer"
[143] "sslcert"                    "sslcertpasswd"
[145] "sslcerttype"                "sslengine"
[147] "sslengine.default"          "sslkey"
[149] "sslkeypasswd"               "sslkeytype"
[151] "sslversion"                 "stderr"
[153] "tcp.nodelay"                "telnetoptions"
[155] "tftp.blksize"               "timecondition"
[157] "timeout"                    "timeout.ms"
[159] "timevalue"                  "transfertext"
[161] "unrestricted.auth"          "upload"
[163] "url"                        "use.ssl"
[165] "useragent"                  "username"
[167] "userpwd"                    "verbose"
[169] "writedata"                  "writefunction"
[171] "writeheader"                "writeinfo"
```

I can see that the `-L` option is not the key here, but rather the `-O` one. I see that `-O` is called `remote-name` but that doesn't seem to match any of the above options by `RCurl`...

```
 curl -Lv http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
*   Trying 10.173.34.79:80...
* Connected to duffel.rail.bio (10.173.34.79) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Date: Wed, 14 Dec 2022 17:24:20 GMT
< Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips PHP/5.6.37
< Last-Modified: Tue, 03 Jan 2017 17:27:19 GMT
< ETag: "2b7-54533faedda24"
< Accept-Ranges: bytes
< Content-Length: 695
< Content-Type: text/html; charset=UTF-8
<
<!DOCTYPE html>
<html>
<head>
<title>Blocked Site</title>
</head>
<body>
<center><img src="http://blocked.jh.edu/logo.png" alt="JH Logo"</center>
<center><h1>Web Page Blocked</h1></center>
<h3><p align="left">
The website you are attempting to visit is blocked by IT@JH security policies due to malicious activity being detected on the remote website. If you believe you have received this message in error, please create a self-service ticket via the myJH portal <a href="https://my.jh.edu">https://my.jh.edu</a> click the helpdesk icon and click "report a problem" or contact the helpdesk directly at 410-955-HELP (4357) or 410-516-HELP (4357).</p>
</h3>
<?php echo 'test' ?>
</body>
</html>

* Connection #0 to host duffel.rail.bio left intact
```

```
% curl -Ov http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 10.173.34.79:80...
* Connected to duffel.rail.bio (10.173.34.79) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Date: Wed, 14 Dec 2022 17:24:27 GMT
< Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips PHP/5.6.37
< Last-Modified: Tue, 03 Jan 2017 17:27:19 GMT
< ETag: "2b7-54533faedda24"
< Accept-Ranges: bytes
< Content-Length: 695
< Content-Type: text/html; charset=UTF-8
<
{ [695 bytes data]
100   695  100   695    0     0  22102      0 --:--:-- --:--:-- --:--:-- 30217
* Connection #0 to host duffel.rail.bio left intact
```


Hmm.... I guess the easiest would be to get duffel back on the approved list by IT @ JH.

**Comment by nellore** [comment]
@lcolladotor Are you on the Hopkins network rn?

**Comment by lcolladotor** [comment]
Yup (at LIBD)

**Comment by nellore** [comment]
@lcolladotor I suspect that's the issue ... you are blocked for whatever reason. Try proxying with HMA or another VPN and see

**Comment by nellore** [comment]
That is, I think this particular problem doesn't affect users outside Hopkins

**Comment by lcolladotor** [comment]
Hmm, switching to my phone as a WiFi hotspot it does work:


```R
> RCurl::url.exists("https://duffel.rail.bio/recount3/human/homes_index", .opts = list(verbose = TRUE))
*   Trying 2606:4700::6810:f34e:443...
* Connected to duffel.rail.bio (2606:4700::6810:f34e) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=duffel.rail.bio
*  start date: Oct 20 00:00:00 2022 GMT
*  expire date: Oct 20 23:59:59 2023 GMT
*  subjectAltName: host "duffel.rail.bio" matched cert's "duffel.rail.bio"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: HEAD]
* h2h3 [:path: /recount3/human/homes_index]
* h2h3 [:scheme: https]
* h2h3 [:authority: duffel.rail.bio]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x124cf7e00)
> HEAD /recount3/human/homes_index HTTP/2
Host: duffel.rail.bio
accept: */*

* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 302
< date: Wed, 14 Dec 2022 17:42:22 GMT
< content-type: text/html; charset=utf-8
< location: https://sciserver.org/public-data/recount3/data/human/homes_index
< x-do-app-origin: 607188ed-e69e-11ec-b1dc-0c42a19a82a7
< cache-control: private
< x-do-orig-status: 302
< cf-cache-status: MISS
< server: cloudflare
< cf-ray: 7798bc55b9038296-IAD
<
* Connection #0 to host duffel.rail.bio left intact
* Issue another request to this URL: 'https://sciserver.org/public-data/recount3/data/human/homes_index'
*   Trying 128.220.233.141:443...
* Connected to sciserver.org (128.220.233.141) port 443 (#1)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server did not agree on a protocol. Uses default.
* Server certificate:
*  subject: C=US; ST=Maryland; O=Johns Hopkins University; CN=sciserver.org
*  start date: Aug 30 00:00:00 2022 GMT
*  expire date: Sep 30 23:59:59 2023 GMT
*  subjectAltName: host "sciserver.org" matched cert's "sciserver.org"
*  issuer: C=US; ST=MI; L=Ann Arbor; O=Internet2; OU=InCommon; CN=InCommon RSA Server CA
*  SSL certificate verify ok.
> HEAD /public-data/recount3/data/human/homes_index HTTP/1.1
Host: sciserver.org
Accept: */*

* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Wed, 14 Dec 2022 17:42:31 GMT
< Server: Apache/2.4.6 (CentOS) PHP/7.3.10
< Last-Modified: Tue, 01 Mar 2022 17:44:39 GMT
< ETag: "35-5d92bb8204c36"
< Accept-Ranges: bytes
< Content-Length: 53
< Cache-Control: max-age=2592000
< Expires: Fri, 13 Jan 2023 17:42:31 GMT
< Set-Cookie: SERVERID=hhpc011; path=/
< Cache-control: private
<
* Connection #1 to host sciserver.org left intact
[1] TRUE
```


Also switching to JHGuestnet


```R
> RCurl::url.exists("https://duffel.rail.bio/recount3/human/homes_index", .opts = list(verbose = TRUE))
*   Trying 104.16.243.78:443...
* Connected to duffel.rail.bio (104.16.243.78) port 443 (#0)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=duffel.rail.bio
*  start date: Oct 20 00:00:00 2022 GMT
*  expire date: Oct 20 23:59:59 2023 GMT
*  subjectAltName: host "duffel.rail.bio" matched cert's "duffel.rail.bio"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: HEAD]
* h2h3 [:path: /recount3/human/homes_index]
* h2h3 [:scheme: https]
* h2h3 [:authority: duffel.rail.bio]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x154fc6e00)
> HEAD /recount3/human/homes_index HTTP/2
Host: duffel.rail.bio
accept: */*

* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 302
< date: Wed, 14 Dec 2022 17:44:08 GMT
< content-type: text/html; charset=utf-8
< location: https://sciserver.org/public-data/recount3/data/human/homes_index
< x-do-app-origin: 607188ed-e69e-11ec-b1dc-0c42a19a82a7
< cache-control: private
< x-do-orig-status: 302
< cf-cache-status: MISS
< server: cloudflare
< cf-ray: 7798beea7a8905fc-IAD
<
* Connection #0 to host duffel.rail.bio left intact
* Issue another request to this URL: 'https://sciserver.org/public-data/recount3/data/human/homes_index'
*   Trying 128.220.233.141:443...
* Connected to sciserver.org (128.220.233.141) port 443 (#1)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server did not agree on a protocol. Uses default.
* Server certificate:
*  subject: C=US; ST=Maryland; O=Johns Hopkins University; CN=sciserver.org
*  start date: Aug 30 00:00:00 2022 GMT
*  expire date: Sep 30 23:59:59 2023 GMT
*  subjectAltName: host "sciserver.org" matched cert's "sciserver.org"
*  issuer: C=US; ST=MI; L=Ann Arbor; O=Internet2; OU=InCommon; CN=InCommon RSA Server CA
*  SSL certificate verify ok.
> HEAD /public-data/recount3/data/human/homes_index HTTP/1.1
Host: sciserver.org
Accept: */*

* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Wed, 14 Dec 2022 17:44:08 GMT
< Server: Apache/2.4.6 (CentOS) PHP/7.3.10
< Last-Modified: Tue, 01 Mar 2022 17:44:39 GMT
< ETag: "35-5d92bb8204c36"
< Accept-Ranges: bytes
< Content-Length: 53
< Cache-Control: max-age=2592000
< Expires: Fri, 13 Jan 2023 17:44:08 GMT
< Set-Cookie: SERVERID=hhpc011; path=/
< Cache-control: private
<
* Connection #1 to host sciserver.org left intact
[1] TRUE
```

**Comment by nellore** [comment]
Possibly it has to do with duffel.rail.bio as referer triggering a block at Hopkins. I do not think the `-LOv` actually worked for you; you got a 404.

**Comment by nellore** [comment]
Probably the same security warning ... but you dumped to file rather than screen

**Comment by lcolladotor** [comment]
Ahh, you are right!

```zcsh
% curl -LOv http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 10.173.34.79:80...
* Connected to duffel.rail.bio (10.173.34.79) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Date: Wed, 14 Dec 2022 17:55:20 GMT
< Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips PHP/5.6.37
< Last-Modified: Tue, 03 Jan 2017 17:27:19 GMT
< ETag: "2b7-54533faedda24"
< Accept-Ranges: bytes
< Content-Length: 695
< Content-Type: text/html; charset=UTF-8
<
{ [695 bytes data]
100   695  100   695    0     0   2107      0 --:--:-- --:--:-- --:--:--  2249
* Connection #0 to host duffel.rail.bio left intact
% more human.gene_sums.G026.gtf.gz
<!DOCTYPE html>
<html>
<head>
<title>Blocked Site</title>
</head>
<body>
<center><img src="http://blocked.jh.edu/logo.png" alt="JH Logo"</center>
<center><h1>Web Page Blocked</h1></center>
<h3><p align="left">
The website you are attempting to visit is blocked by IT@JH security policies due to malicious activity being detected on the remote website. If you believe you have received this message in error, please create a self-service ticket via the myJH portal <a href="https://my.jh.edu">https://my.jh.edu</a> click the helpdesk icon and click "report a problem" or contact the helpdesk directly at 410-955-HELP (4357) or 410-516-HELP (4357).</p>
</h3>
<?php echo 'test' ?>
</body>
</html>
```

**Comment by nellore** [comment]
This is likely the same issue Ben was having this morning. Sent an email
about whitelisting duffel this morning (before knowing it was Hopkins
network-specific) — maybe you can follow up with the info on how this is
happening on jhaccess and not the guest network? Could be helpful.

**Comment by lcolladotor** [comment]
<img width="1188" alt="Screenshot 2022-12-14 at 1 02 39 PM" src="https://user-images.githubusercontent.com/2288213/207672153-05f71eda-e91b-4374-8d32-421f9dba66bb.png">

I created a ticket at jh.edu a few minutes ago, before you noticed that `curl -LOv` didn't work for me on the Hopkins network.

If you cc me on the email to IT I can add the info about JHGuestnet + phone hotspot.

**Comment by nellore** [comment]
email sent!

**Comment by ndaniel** [comment]
Here is more info:
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy
```

```
$ curl -LOv http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 104.16.243.78:80...
* Connected to duffel.rail.bio (104.16.243.78) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 301 Moved Permanently
< Date: Thu, 15 Dec 2022 09:34:26 GMT
< Transfer-Encoding: chunked
< Connection: keep-alive
< Cache-Control: max-age=3600
< Expires: Thu, 15 Dec 2022 10:34:26 GMT
< Location: https://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
< Server: cloudflare
< CF-RAY: 779e2ef62825c7de-TLL
< 
* Ignoring the response-body
{ [5 bytes data]
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Connection #0 to host duffel.rail.bio left intact
* Clear auth, redirects to port from 80 to 443
* Issue another request to this URL: 'https://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz'
*   Trying 104.16.243.78:443...
* Connected to duffel.rail.bio (104.16.243.78) port 443 (#1)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.2 (IN), TLS header, Certificate Status (22):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* TLSv1.2 (IN), TLS header, Finished (20):
{ [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
{ [19 bytes data]
* TLSv1.3 (IN), TLS handshake, Certificate (11):
{ [2294 bytes data]
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
{ [79 bytes data]
* TLSv1.3 (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* TLSv1.2 (OUT), TLS header, Finished (20):
} [5 bytes data]
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=duffel.rail.bio
*  start date: Oct 20 00:00:00 2022 GMT
*  expire date: Oct 20 23:59:59 2023 GMT
*  subjectAltName: host "duffel.rail.bio" matched cert's "duffel.rail.bio"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* Using Stream ID: 1 (easy handle 0x56353d8b9e80)
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/2
> Host: duffel.rail.bio
> user-agent: curl/7.81.0
> accept: */*
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [230 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [230 bytes data]
* old SSL session ID is stale, removing
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
< HTTP/2 302 
< date: Thu, 15 Dec 2022 09:34:26 GMT
< content-type: text/html; charset=utf-8
< location: https://sciserver.org/public-data/recount3/data/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
< x-do-app-origin: 607188ed-e69e-11ec-b1dc-0c42a19a82a7
< cache-control: private
< x-do-orig-status: 302
< cf-cache-status: MISS
< server: cloudflare
< cf-ray: 779e2ef70a32c7e7-TLL
< 
* Ignoring the response-body
{ [413 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
100   413    0   413    0     0    980      0 --:--:-- --:--:-- --:--:--     0
* Connection #1 to host duffel.rail.bio left intact
* Issue another request to this URL: 'https://sciserver.org/public-data/recount3/data/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz'
*   Trying 128.220.233.141:443...
* Connected to sciserver.org (128.220.233.141) port 443 (#2)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.2 (IN), TLS header, Certificate Status (22):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [93 bytes data]
* TLSv1.2 (IN), TLS header, Certificate Status (22):
{ [5 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [5745 bytes data]
* TLSv1.2 (IN), TLS header, Certificate Status (22):
{ [5 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [333 bytes data]
* TLSv1.2 (OUT), TLS header, Unknown (21):
} [5 bytes data]
* TLSv1.2 (OUT), TLS alert, handshake failure (552):
} [2 bytes data]
* error:0A000172:SSL routines::wrong signature type
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Closing connection 2
curl: (35) error:0A000172:SSL routines::wrong signature type
```

**Comment by nellore** [comment]
Thanks for the extra info. We believe the sciserver admins will address this issue in the near future, but for now, glad you have workaround #3!

**Comment by lcolladotor** [comment]
Looks like the JH @ IT ban I had is gone now:

```bash
% wget http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
--2022-12-16 12:35:30--  http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
Resolving duffel.rail.bio (duffel.rail.bio)... 104.16.244.78, 104.16.243.78
Connecting to duffel.rail.bio (duffel.rail.bio)|104.16.244.78|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz [following]
--2022-12-16 12:35:30--  https://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
Connecting to duffel.rail.bio (duffel.rail.bio)|104.16.244.78|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://sciserver.org/public-data/recount3/data/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz [following]
--2022-12-16 12:35:31--  https://sciserver.org/public-data/recount3/data/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
Resolving sciserver.org (sciserver.org)... 128.220.233.141
Connecting to sciserver.org (sciserver.org)|128.220.233.141|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1842619 (1.8M) [application/x-gzip]
Saving to: ‘human.gene_sums.G026.gtf.gz’

human.gene_sums.G026.gtf.gz       100%[==========================================================>]   1.76M  1.69MB/s    in 1.0s

2022-12-16 12:35:32 (1.69 MB/s) - ‘human.gene_sums.G026.gtf.gz’ saved [1842619/1842619]
```

```bash
% curl -LOv http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 104.16.243.78:80...
* Connected to duffel.rail.bio (104.16.243.78) port 80 (#0)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: duffel.rail.bio
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 301 Moved Permanently
< Date: Fri, 16 Dec 2022 17:36:38 GMT
< Transfer-Encoding: chunked
< Connection: keep-alive
< Cache-Control: max-age=3600
< Expires: Fri, 16 Dec 2022 18:36:38 GMT
< Location: https://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
< Server: cloudflare
< CF-RAY: 77a92eae7b0c5716-IAD
<
* Ignoring the response-body
{ [5 bytes data]
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Connection #0 to host duffel.rail.bio left intact
* Clear auth, redirects to port from 80 to 443
* Issue another request to this URL: 'https://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz'
*   Trying 104.16.243.78:443...
* Connected to duffel.rail.bio (104.16.243.78) port 443 (#1)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (OUT), TLS handshake, Client hello (1):
} [320 bytes data]
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [19 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [2294 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [78 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [36 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [36 bytes data]
* SSL connection using TLSv1.3 / AEAD-CHACHA20-POLY1305-SHA256
* ALPN: server accepted h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=duffel.rail.bio
*  start date: Oct 20 00:00:00 2022 GMT
*  expire date: Oct 20 23:59:59 2023 GMT
*  subjectAltName: host "duffel.rail.bio" matched cert's "duffel.rail.bio"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* h2h3 [:method: GET]
* h2h3 [:path: /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz]
* h2h3 [:scheme: https]
* h2h3 [:authority: duffel.rail.bio]
* h2h3 [user-agent: curl/7.84.0]
* h2h3 [accept: */*]
* Using Stream ID: 1 (easy handle 0x15780c600)
> GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/2
> Host: duffel.rail.bio
> user-agent: curl/7.84.0
> accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 302
< date: Fri, 16 Dec 2022 17:36:38 GMT
< content-type: text/html; charset=utf-8
< location: https://sciserver.org/public-data/recount3/data/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
< x-do-app-origin: 607188ed-e69e-11ec-b1dc-0c42a19a82a7
< cache-control: private
< x-do-orig-status: 302
< cf-cache-status: MISS
< server: cloudflare
< cf-ray: 77a92eaf0f37801d-IAD
<
* Ignoring the response-body
{ [413 bytes data]
100   413    0   413    0     0   1428      0 --:--:-- --:--:-- --:--:--  1428
* Connection #1 to host duffel.rail.bio left intact
* Issue another request to this URL: 'https://sciserver.org/public-data/recount3/data/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz'
*   Trying 128.220.233.141:443...
* Connected to sciserver.org (128.220.233.141) port 443 (#2)
* ALPN: offers h2
* ALPN: offers http/1.1
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (OUT), TLS handshake, Client hello (1):
} [318 bytes data]
* (304) (IN), TLS handshake, Server hello (2):
{ [93 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [5745 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [333 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [70 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
{ [1 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN: server did not agree on a protocol. Uses default.
* Server certificate:
*  subject: C=US; ST=Maryland; O=Johns Hopkins University; CN=sciserver.org
*  start date: Aug 30 00:00:00 2022 GMT
*  expire date: Sep 30 23:59:59 2023 GMT
*  subjectAltName: host "sciserver.org" matched cert's "sciserver.org"
*  issuer: C=US; ST=MI; L=Ann Arbor; O=Internet2; OU=InCommon; CN=InCommon RSA Server CA
*  SSL certificate verify ok.
> GET /public-data/recount3/data/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> Host: sciserver.org
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Fri, 16 Dec 2022 17:36:38 GMT
< Server: Apache/2.4.6 (CentOS) PHP/7.3.10
< Last-Modified: Wed, 29 Apr 2020 19:13:37 GMT
< ETag: "1c1dbb-5a472bebf7a40"
< Accept-Ranges: bytes
< Content-Length: 1842619
< Content-Type: application/x-gzip
< Set-Cookie: SERVERID=hhpc009; path=/
< Cache-control: private
<
{ [14206 bytes data]
100 1799k  100 1799k    0     0  2156k      0 --:--:-- --:--:-- --:--:-- 5189k
* Connection #2 to host sciserver.org left intact
```

If so, then the only issues remaining would be:

*  the TLS 1.3 update by SciServer admins, right? The one affecting @ndaniel.
* fixing https://jhubiostatistics.shinyapps.io/recount3-study-explorer/


Right?

The shiny app is running into the error that started this: `Error in : 'recount3_url' is not a valid supported URL since it's missing the URL/<organism>/homes_index text file or 'recount3_url' is not an existing directory in your file system.`. So maybe shinyapps.io got blocked by IT @ JH.

```
2022-12-16T17:38:46.562876+00:00 shinyapps[3529357]: Warning: Error in : 'recount3_url' is not a valid supported URL since it's missing the URL/<organism>/homes_index text file or 'recount3_url' is not an existing directory in your file system.
2022-12-16T17:38:46.571232+00:00 shinyapps[3529357]:   120: stop
2022-12-16T17:38:46.571249+00:00 shinyapps[3529357]:   119: project_homes
2022-12-16T17:38:46.571253+00:00 shinyapps[3529357]:   115: recount3::locate_url
2022-12-16T17:38:46.571256+00:00 shinyapps[3529357]:   114: .f
2022-12-16T17:38:46.571259+00:00 shinyapps[3529357]:   113: purrr::map
2022-12-16T17:38:46.571263+00:00 shinyapps[3529357]:   111: renderUI [/srv/connect/apps/recount3-study-explorer/server.R#96]
2022-12-16T17:38:46.571272+00:00 shinyapps[3529357]:   110: func
2022-12-16T17:38:46.571276+00:00 shinyapps[3529357]:    97: renderFunc
2022-12-16T17:38:46.571278+00:00 shinyapps[3529357]:    96: output$raw_file_links
2022-12-16T17:38:46.571287+00:00 shinyapps[3529357]:    15: <Anonymous>
2022-12-16T17:38:46.571290+00:00 shinyapps[3529357]:    13: fn
2022-12-16T17:38:46.571293+00:00 shinyapps[3529357]:     8: retry
2022-12-16T17:38:46.571296+00:00 shinyapps[3529357]:     7: connect$retryingStartServer
2022-12-16T17:38:46.571302+00:00 shinyapps[3529357]:     6: eval
2022-12-16T17:38:46.571306+00:00 shinyapps[3529357]:     5: eval
2022-12-16T17:38:46.848558+00:00 shinyapps[3529357]: Warning: Error in : 'recount3_url' is not a valid supported URL since it's missing the URL/<organism>/homes_index text file or 'recount3_url' is not an existing directory in your file system.
2022-12-16T17:38:46.859277+00:00 shinyapps[3529357]:   120: stop
2022-12-16T17:38:46.859316+00:00 shinyapps[3529357]:   119: project_homes
2022-12-16T17:38:46.859322+00:00 shinyapps[3529357]:   115: recount3::locate_url
2022-12-16T17:38:46.859340+00:00 shinyapps[3529357]:   114: .f
2022-12-16T17:38:46.859346+00:00 shinyapps[3529357]:   113: purrr::map
2022-12-16T17:38:46.859350+00:00 shinyapps[3529357]:   111: renderUI [/srv/connect/apps/recount3-study-explorer/server.R#96]
2022-12-16T17:38:46.859356+00:00 shinyapps[3529357]:   110: func
2022-12-16T17:38:46.859361+00:00 shinyapps[3529357]:    97: renderFunc
2022-12-16T17:38:46.859369+00:00 shinyapps[3529357]:    96: output$raw_file_links
2022-12-16T17:38:46.859372+00:00 shinyapps[3529357]:    15: <Anonymous>
2022-12-16T17:38:46.859379+00:00 shinyapps[3529357]:    13: fn
2022-12-16T17:38:46.859382+00:00 shinyapps[3529357]:     8: retry
2022-12-16T17:38:46.859385+00:00 shinyapps[3529357]:     7: connect$retryingStartServer
2022-12-16T17:38:46.859388+00:00 shinyapps[3529357]:     6: eval
2022-12-16T17:38:46.859391+00:00 shinyapps[3529357]:     5: eval
```

**Comment by lcolladotor** [comment]
I just created a new ticket with IT at JHU.

Here's part of the message I wrote:

> To whom it may concern from the IT @ JHU Network Security team,
> 
> As part of two of our research projects (recount2 and recount3), we generated about 200 TB of data that is hosted by IDIES at JHU. Admins at IDIES recently changed the data host from http://idies.jhu.edu/ to https://sciserver.org/public-data/. This broke data access for many users who have been asking us about this at https://github.com/LieberInstitute/recount3/issues/29 and via emails. We use duffel.rail.bio, maintained by one of our team members, to redirect URLs as we anticipated needing to change the data host at some point. While this end data location was easy for us to fix (see https://github.com/nellore/digitalocean-duffel/commit/b6a091b92bff7e25166e07b879234ae3b982657c), IT at @ JH is automatically blocking access to the new data location for myself on my JHU laptop (on the JHaccess and hopkins WiFi networks, not on JHGuestnet) as well as other users including our interactive website https://jhubiostatistics.shinyapps.io/recount3-study-explorer/ (part of http://rna.recount.bio/).  The easiest way to check is with the URL https://duffel.rail.bio/recount3/human/homes_index which right now redirects to https://sciserver.org/public-data/recount3/data/human/homes_index. It seems that traffic from duffel.rail.bio is being blocked by IT at JH to https://sciserver.org/public-data (which is hosted by IDIES on the JHU network). See for example this (that I ran on the JHaccess network):
> 
> % curl -v http://duffel.rail.bio/recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz
> *   Trying 10.173.34.79:80...
> * Connected to duffel.rail.bio (10.173.34.79) port 80 (#0)
> > GET /recount3/human/annotations/gene_sums/human.gene_sums.G026.gtf.gz HTTP/1.1
> > Host: duffel.rail.bio
> > User-Agent: curl/7.84.0
> > Accept: */*
> >
> * Mark bundle as not supporting multiuse
> < HTTP/1.1 404 Not Found
> < Date: Fri, 16 Dec 2022 18:47:07 GMT
> < Server: Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.2k-fips PHP/5.6.37
> < Last-Modified: Tue, 03 Jan 2017 17:27:19 GMT
> < ETag: "2b7-54533faedda24"
> < Accept-Ranges: bytes
> < Content-Length: 695
> < Content-Type: text/html; charset=UTF-8
> <
> <!DOCTYPE html>
> <html>
> <head>
> <title>Blocked Site</title>
> </head>
> <body>
> <center><img src="http://blocked.jh.edu/logo.png" alt="JH Logo"</center>
> <center><h1>Web Page Blocked</h1></center>
> <h3><p align="left">
> The website you are attempting to visit is blocked by IT@JH security policies due to malicious activity being detected on the remote website. If you believe you have received this message in error, please create a self-service ticket via the myJH portal <a href="https://my.jh.edu">https://my.jh.edu</a> click the helpdesk icon and click "report a problem" or contact the helpdesk directly at 410-955-HELP (4357) or 410-516-HELP (4357).</p>
> </h3>
> <?php echo 'test' ?>
> </body>
> </html>
> 
> * Connection #0 to host duffel.rail.bio left intact
> 
> 
> 
>  I'm a bit confused on what are the exact steps that need to be taken to restore full access to the data we have hosted by IDIES, and whether IDIES admins or central IT at JHU need to change something to restore access from duffel.rail.bio to the data hosted by IDIES (now at sciserver.org). As noted at https://github.com/LieberInstitute/recount3/issues/29#issuecomment-1355290006 we believe that there are 2 issues:
> 
> * SciServer admins need to support TLS 1.3 on https://sciserver.org/public-data .
> * IT @ JH likely need to whitelist traffic from duffel.rail.bio

**Comment by lcolladotor** [comment]
Hi everyone,

So, IT @ JHU lifted the block for me and I can now access https://duffel.rail.bio/recount3/human/homes_index from a JHU WiFi. However, even if I try to bypass duffel at https://jhubiostatistics.shinyapps.io/recount3-study-explorer/ (with https://github.com/LieberInstitute/recount3-docs/commit/d4ff18098b268a1bdb695c2bc2a3782e5e0d808d) it's still being blocked.

Independently, another user might also running into a block by IT @ JHU. I asked that user to try `curl -v` to see if the block message appears for them. More details at https://support.bioconductor.org/p/9148519/.

We also heard from IDIES that they are going to be doing some more changes that will hopefully resolve the migration of the data host, but I'm guessing this will take some time.

Best,
Leo

**Comment by lcolladotor** [comment]
Hi everyone,

https://jhubiostatistics.shinyapps.io/recount3-study-explorer/ is working again now, thanks to behind the scenes work by @nellore and @ChristopherWilks.

Thanks!

**Comment by lcolladotor** [comment]
Hi everyone,

I'm closing this issue now that we have a new data host. As for the TLS version 1.2 vs 1.1 access, we could move that to a new issue.

Thanks everyone!

https://twitter.com/BenLangmead/status/1613196179587883008

**Comment by lcolladotor** [comment]
The new data host is now also documented on `recount3` version 1.9.1 https://github.com/LieberInstitute/recount3/commit/6cf18f316123695b6a93c2049ab499b00d6c2acf.

**Comment by bitmman** [comment]
Is the server down again?
`human_projects <- recount3::project_homes("human", recount3_url = "https://sciserver.org/public-data/recount3/data")` reports the following error again: "Error: 'recount3_url' is not a valid supported URL since it's missing the URL/<organism>/homes_index text file or 'recount3_url' is not an existing directory in your file system."

I have updated recount3 to the latest version 1.9.1. 

The SciServer link [https://sciserver.org/public-data/recount3/data](https://sciserver.org/public-data/recount3/data) does not exist.

**Comment by lcolladotor** [comment]
SciServer is making some internal changes, but well, `recount3` (through duffel) actually doesn't use SciServer right now but AWS. So if you don't specify a `recount3_url` it should work. Unless you are on Windows and run into issue with duffel. In that case, use `options(recount3_url = "https://recount-opendata.s3.amazonaws.com/recount3/release")`.
